macros/PG.pl

```diff
@@ -77,27 +77,38 @@ sub DOCUMENT {
 	#FIXME
 	# load java script needed for displayModes
 	if ($envir{displayMode} eq 'HTML_MathJax') {
-		##
-		# The following is used to prevent MathJax from being loaded more than one on a page. (Happened in WW3 library browser)
-		#
-		# If there are any errors replace the line starting "window.MathJax" with the string in $loadScript.
-		#
-	    TEXT(
-		 qq?<script type="text/x-mathjax-config">
-                  MathJax.Hub.Config({
-                     MathMenu: {showContext: true}
-                  });
-                  </script>
-				  <script type="text/javascript"> 
-				  if(!window.MathJax) 
-				  (function () {
-  					var script = document.createElement("script");
-  					script.type = "text/javascript";
-  					script.src  = "$envir{MathJaxURL}";
-  					document.getElementsByTagName("head")[0].appendChild(script);
-					})();                
-                  </script>?."\n");
-        } elsif ($envir{displayMode} eq 'HTML_jsMath') {
+		TEXT(
+			qq?<script type="text/javascript">
+				if (!window.MathJax) 
+				{
+					window.MathJax = {
+						tex: {
+							autoload: {
+								color: [],
+								colorV2: ['color']
+							},
+							packages: {'[+]': ['noerrors']}
+						},
+						loader: {
+							load: ['input/asciimath', '[tex]/noerrors']
+						},
+						startup: {
+							ready: function() {
+								var AM = MathJax.InputJax.AsciiMath.AM;
+								for (var i = 0; i < AM.symbols.length; i++) {
+									if (AM.symbols[i].input == '**') {
+										AM.symbols[i] = {
+											input: "**", tag: "msup", output: "^", tex: null, ttype: AM.TOKEN.INFIX
+										};
+									}
+								}
+								return MathJax.startup.defaultReady()
+							}
+						}
+					};
```

Comment:
Since this configuration gets used in two places (here and in `Instructor/ShowAnswers.pm`), I'm wondering if this could be put into a separate .js file that is loaded here and there rather than repeating the same code.  That way, the code in both places won't get our of sync.

Reply:
That sounds good.

Reply:
If you do make it a separate file, you may want to load it in the template along with the other scripts, as it needs to come before the script that loads `tex-chtml.js` (otherwise setting `MathJax` here will overwrite the `MathJax` from that file).  So if you use `defer` on loading this file followed by `defer` on loading `tex-chtml.js` (with loading polyfil with `defer` before both), you should be OK.

Reply:
I didn't add the `defer` attribute to the polyfill javascript.  So that should have it to then?

Reply:
Would the `async` attribute be better for the polyfill.io script?  That shouldn't need to wait for the DOM content to be loaded.

Reply:
I think that might allow it to run out of order.  For example, if polyfill took a long time to get to the page, the since it is allowed to run whenever it arrives, it might be that the `defer` items are run first.  I don't know for sure about that, but I do know that `defer` requires the order to be preserved, while `async` does not.

Reply:
Okay.  I will leave it on `defer` then.